docs/dev_setup.md

```diff
@@ -30,101 +28,43 @@ echo 'eval "$(direnv hook bash)"' >> ~/.bashrc  # if you use bash
 eval "$(direnv hook bash)"  # add to current shell
 
-# Add PDM support to direnv
+# Add uv support to direnv
 mkdir -p ~/.config/direnv/
 cp .devcontainer/direnvrc ~/.config/direnv/direnvrc
 ```
+
 When you enter the project folder, direnv will ask you to allow the `.envrc` file to be loaded.
 Allow it by running `direnv allow`.
 
+### [uv](https://docs.astral.sh/uv/)
 
-### [Pyenv](https://github.com/pyenv/pyenv)
-
-For installing and managing (multiple) python versions.
-You may want to use the system's version and not install pyenv.
-
-Otherwise, follow these instructions.
-
-1. [Install Python build dependencies](https://github.com/pyenv/pyenv?tab=readme-ov-file#install-python-build-dependencies):
-    ```bash
-    sudo apt update && sudo apt install build-essential libssl-dev zlib1g-dev \
-        libbz2-dev libreadline-dev libsqlite3-dev curl \
-        libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
-    ```
-2. [Install pyenv](https://github.com/pyenv/pyenv?tab=readme-ov-file#automatic-installer)
-    ```bash
-    curl https://pyenv.run | bash
-    ```
-
-3. [Set up your shell environment for Pyenv](https://github.com/pyenv/pyenv?tab=readme-ov-file#set-up-your-shell-environment-for-pyenv)
-    ```bash
-    # add env variables to your bash .profile (if you use bash)
-    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
-    echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
-    echo 'eval "$(pyenv init --path)"' >> ~/.profile
-
-    # add shell integration to bash shell
-    echo 'eval "$(pyenv init -)"' >> ~/.bashrc
-
-    # add env variables to zsh's .zprofile (if you use zsh)
-    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.zprofile
-    echo '[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.zprofile
-    echo 'eval "$(pyenv init --path)"' >> ~/.zprofile
-
-    # add shell integration to zsh shell
-    echo 'eval "$(pyenv init -)"' >> ~/.zshrc
-    ```
-
-4. [Install python versions](https://github.com/pyenv/pyenv?tab=readme-ov-file#install-additional-python-versions)
-    ```bash
-    pyenv install 3.10
-    pyenv global 3.10
-    ```
-
-
-### [Pipx](https://pipxproject.github.io/pipx/)
-
-For installing python packages in isolated environments.
-
-```bash
-sudo apt update && sudo apt install pipx
-pipx ensurepath
-```
-
-
-### [PDM](https://pdm-project.org/)
 To manage package dependencies and virtual environments.
 
 ```bash
-pipx install pdm
+curl -LsSf https://astral.sh/uv/install.sh | sh
 ```
 
-
 ### [Pre-commit](https://pre-commit.com/)
-To run code formatting and linting before committing.
-
-```bash
-pipx install pre-commit
-```
 
+To run code formatting and linting before committing.
 In your projects, you should enable pre-commit hooks by running:
+
 ```bash
-pre-commit install --install-hooks
+uvx --with pre-commit-uv -- pre-commit install --install-hooks
 ```
```

Comment:
I would argue it is better to install pre-commit in the system rather than per project? Don't you agree? What I don't like about uvx is that you may end up with outdated cached versions of pre-commit different in each project...
```suggestion
uv tool install pre-commit --with pre-commit-uv
pre-commit install --install-hooks
```